scripts/todesktop/beforeInstall.js

```diff
@@ -19,5 +19,8 @@ module.exports = async ({ pkgJsonPath, pkgJson, appDir, hookName }) => {
         // Change stdio to get back the logs if there are issues.
         const result1 = spawnSync('curl' ,['-s', 'https://www.python.org/ftp/python/3.12.7/python-3.12.7-amd64.exe'],{shell:true,stdio: 'ignore'}).toString();
         const result2 = spawnSync('python-3.12.7-amd64.exe', ['/quiet', 'InstallAllUsers=1','PrependPath=1', 'Include_test=0'],{shell:true,stdio: 'ignore'}).toString();
+
+        // Download Visual C Redistributable depedency for Pytorch (-L sets to follow redirects) into %TEMP% (will be picked up by installer.nsh)
+        const result3 = spawnSync('curl', ['-s', '-L', 'https://aka.ms/vs/16/release/vc_redist.x64.exe', '-o', '%TEMP%/vc_redist.x64.exe'],{shell:true,stdio: 'ignore'}).toString();
```

Comment:
Question on this, what does the TEMP directory mean here? 

Ideally I want to move most things outside of beforeInstall because ToDesktop logs are not very reliable. For example for uv we pre-download everything on Github actions and upload everything to ToDestkop.

Reply:
Since I'm including the redistributable in the installer I need to point NSIS file to location where it can grab it; unfortunately the location of installer build is ephemeral and I need full path so I opted to use official Windows temporary directory to download the installer into (see `beforeInstall.js` parameters `'-o', '%TEMP%/vc_redist.x64.exe'`) and then NSIS can pick it up from there.